cmd/contour/shutdownmanager.go

```diff
@@ -98,17 +101,44 @@ func (s *shutdownmanagerContext) healthzHandler(w http.ResponseWriter, r *http.R
 // file to understand if it is safe to terminate. The file-based approach is used since the process in which
 // the kubelet calls the shutdown command is different than the HTTP request from Envoy to /shutdown
 func (s *shutdownmanagerContext) shutdownReadyHandler(w http.ResponseWriter, r *http.Request) {
+	// For testing, the filepath and check period can be overridden in the context
+	// Otherwise, use the module constants
+	var signalFilepath string
+	if s.shutdownReadyFile != "" {
+		signalFilepath = s.shutdownReadyFile
+	} else {
+		signalFilepath = shutdownReadyFile
+	}
+	var signalFileCheckDuration time.Duration
+	if s.shutdownReadyCheckInterval != 0 {
+		signalFileCheckDuration = s.shutdownReadyCheckInterval
+	} else {
+		signalFileCheckDuration = shutdownReadyCheckInterval
+	}
```

Comment:
Not too happy with the location of this logic... maybe the defaults assignment should move to `newShutdownManagerContext()`.

Reply:
Yeah, I'd probably assign the default values to these fields in `newShutdownManagerContext()`, then you can just override them as needed.

Reply:
Done!